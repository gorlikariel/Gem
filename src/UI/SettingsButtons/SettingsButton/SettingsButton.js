import React from "react";
import { Link } from "react-router-dom";
import WideButton from "../../WideButton/WideButton";
import { Typography } from "@material-ui/core";
import AccountIcon from "../../Icons/AccountIcon/AccountIcon";
import ClockIcon from "../../Icons/ClockIcon/ClockIcon";
import PillIcon from "../../Icons/PillIcon/PillIcon";
import * as theme from "../../theme/theme";
import RightArrow from "../../Icons/RightArrow/RightArrow";
const SettingsButton = props => {
  const { bgColor, color, to, children, icon } = props;
  const clicked = () => {
    console.log("FUUUCK");
  };
  const rightArrow = <RightArrow style={{ paddingRight: "12px" }} />;
  const styles = {
    buttonIcon: {
      paddingLeft: "20px"
    },
    logoutButton: {
      width: "24px",
      paddingLeft: "20px"
    },
    navLink: {
      textDecoration: "none",
      color: "inherit"
    }
  };
  const icons = {
    accountIcon: <AccountIcon style={styles.buttonIcon} />,
    alarmIcon: <ClockIcon style={styles.buttonIcon} />,
    pillIcon: <PillIcon style={styles.buttonIcon} />
  };

  return (
    <Link style={{ textDecoration: "none" }} to={to}>
      <WideButton bgColor={theme[bgColor]} color={theme[color]}>
        {icons[icon]}
        <Typography
          style={{ paddingLeft: to === "/" ? "54px" : "12px" }}
          variant="display3"
          color="inherit"
        >
          {children}
        </Typography>
        <div style={{ marginLeft: "auto" }}>
          {to === "/" ? null : rightArrow}
        </div>
      </WideButton>
    </Link>
  );
};
export default SettingsButton;
