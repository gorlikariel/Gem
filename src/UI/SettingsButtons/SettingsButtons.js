import React from "react";
import WideButton from "../BasicButton/WideButton";
import * as theme from "../theme/theme";
import AccountIcon from "../Icons/AccountIcon/AccountIcon";
import { Typography } from "@material-ui/core";
import RightArrow from "../Icons/RightArrow/RightArrow";
import { Link } from "react-router-dom";
import ClockIcon from "../Icons/ClockIcon/ClockIcon";
import PillIcon from "../Icons/PillIcon/PillIcon";

const SettingsButtons = props => {
  let rightArrow = <RightArrow style={{ paddingRight: "12px" }} />;
  const styles = {
    buttonIcon: {
      paddingLeft: "20px"
    },
    logoutButton: {
      width: "24px",
      paddingLeft: "20px"
    },
    navLink: {
      textDecoration: "none",
      color: "inherit"
    }
  };
  return (
    <div>
      <Link style={{ textDecoration: "none" }} to="/account-settings">
        <WideButton bgColor="G3" color={theme.PWH}>
          <AccountIcon style={styles.buttonIcon} />
          <Typography
            style={{ paddingLeft: "12px" }}
            variant="display3"
            color="inherit"
          >
            Account Settings
          </Typography>
          <div style={{ marginLeft: "auto" }}>{rightArrow}</div>
        </WideButton>
      </Link>
      <Link style={{ textDecoration: "none" }} to="/alarm-settings">
        <WideButton bgColor="G2" color={theme.PWH}>
          <ClockIcon style={styles.buttonIcon} />
          <Typography
            style={{ paddingLeft: "12px" }}
            variant="display3"
            color="inherit"
          >
            Alarm Settings
          </Typography>
          <div style={{ marginLeft: "auto" }}>{rightArrow}</div>
        </WideButton>
      </Link>
      <Link style={{ textDecoration: "none" }} to="/pill-settings">
        <WideButton bgColor="PG" color={theme.PWH}>
          <PillIcon style={styles.buttonIcon} />
          <Typography
            style={{ paddingLeft: "12px" }}
            variant="display3"
            color="inherit"
          >
            Pill Settings
          </Typography>
          <div style={{ marginLeft: "auto" }}>{rightArrow}</div>
        </WideButton>
      </Link>
      <Link style={{ textDecoration: "none" }} to="/">
        <WideButton bgColor="PWH" color="black">
          <Typography
            style={{ paddingLeft: "54px" }}
            variant="display3"
            color="inherit"
          >
            Log out of {props.username}
          </Typography>
        </WideButton>
      </Link>
    </div>
  );
};

export default SettingsButtons;
