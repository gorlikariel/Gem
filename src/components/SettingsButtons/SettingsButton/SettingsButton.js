import React from 'react';
import { Link } from 'react-router-dom';
import WideButton from '../../WideButton/WideButton';
import { Typography, Collapse } from '@material-ui/core';
import * as theme from '../../../styleguide/theme';
import AccountIcon from '../../../Icons/AccountIcon/AccountIcon';
import RightArrow from '../../../Icons/RightArrow/RightArrow';
import PillIcon from '../../../Icons/PillIcon/PillIcon';
import ClockIcon from '../../../Icons/ClockIcon/ClockIcon';
const SettingsButton = props => {
  const { bgColor, color, to, children, icon } = props;
  const rightArrow = <RightArrow style={{ paddingRight: '12px' }} />;
  const styles = {
    buttonIcon: {
      paddingLeft: '20px'
    },
    logoutButton: {
      width: '24px',
      paddingLeft: '20px'
    }
  };
  const icons = {
    accountIcon: <AccountIcon style={styles.buttonIcon} />,
    alarmIcon: <ClockIcon style={styles.buttonIcon} />,
    pillIcon: <PillIcon style={styles.buttonIcon} />
  };

  return (
    <Collapse in={props.in} timeout={props.timeout}>
      <Link style={{ textDecoration: 'none' }} to={to}>
        <div
          style={{
            borderImage: ' linear-gradient(to left, #743ad5 0%, #d53a9d 100%)',
            borderImageSlice: 1,
            borderWidth: '10px'
          }}
        >
          <WideButton
            onClick={props.onClick}
            bgColor={bgColor}
            color={color}
            onClick={props.onClick}
            noBoxShadow
          >
            {icons[icon]}
            <Typography
              style={{ paddingLeft: to === '/' ? '55px' : '12px' }}
              variant="display3"
              color="inherit"
            >
              {children}
            </Typography>
            <div style={{ marginLeft: 'auto' }}>
              {to === '/' ? null : rightArrow}
            </div>
          </WideButton>
        </div>
      </Link>
    </Collapse>
  );
};
export default SettingsButton;
