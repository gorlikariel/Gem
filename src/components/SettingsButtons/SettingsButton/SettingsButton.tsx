import * as React from 'react';
import { Link } from 'react-router-dom';
import WideButton from '../../WideButton/WideButton';
import { Typography, Collapse } from '@material-ui/core';
import AccountIcon from '../../../Icons/AccountIcon/AccountIcon';
import RightArrow from '../../../Icons/RightArrow/RightArrow';
import PillIcon from '../../../Icons/PillIcon/PillIcon';
import ClockIcon from '../../../Icons/ClockIcon/ClockIcon';
interface SettingsButtonProps {
  bgColor: string;
  color: string;
  to: string;
  icon?: string;
  timeout?: number;
  onClick?: React.EventHandler<any>;
  in?: boolean;
}

const SettingsButton: React.SFC<SettingsButtonProps> = props => {
  const { bgColor, color, to, children, icon, timeout, onClick } = props;
  const rightArrow = <RightArrow style={{ paddingRight: '12px' }} />;
  const styles = {
    buttonIcon: {
      paddingLeft: '20px'
    },
    logoutButton: {
      width: '24px',
      paddingLeft: '20px'
    },
    rightArrow: { marginLeft: 'auto' },
    buttonText: { paddingLeft: to === '/' ? '55px' : '12px' },
    link: { textDecoration: 'none' }
  };
  const icons = {
    accountIcon: <AccountIcon style={styles.buttonIcon} />,
    alarmIcon: <ClockIcon style={styles.buttonIcon} />,
    pillIcon: <PillIcon style={styles.buttonIcon} />
  };

  return (
    <Collapse in={props.in} timeout={timeout}>
      <Link style={styles.link} to={to}>
        <WideButton onClick={onClick} bgColor={bgColor} color={color}>
          {icon ? icons[icon] : null}
          <Typography
            style={styles.buttonText}
            variant="display3"
            color="inherit"
          >
            {children}
          </Typography>
          <div style={styles.rightArrow}>{to === '/' ? null : rightArrow}</div>
        </WideButton>
      </Link>
    </Collapse>
  );
};
export default SettingsButton;
