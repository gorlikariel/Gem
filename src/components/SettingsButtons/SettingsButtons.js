import React from "react";
import SettingsButton from "./SettingsButton/SettingsButton";
const SettingsButtons = props => (
  <React.Fragment>
    <SettingsButton
      icon="accountIcon"
      bgColor="G3"
      color="PWH"
      to="/account-settings"
    >
      Account Settings
    </SettingsButton>
    <SettingsButton
      icon="alarmIcon"
      bgColor="G2"
      color="PWH"
      to="/alarm-settings"
    >
      Alarm Settings
    </SettingsButton>
    <SettingsButton
      icon="pillIcon"
      bgColor="PG"
      color="PWH"
      to="/pill-settings"
    >
      Pill Settings
    </SettingsButton>
    <SettingsButton
      onClick={() => console.log("FUCK")}
      bgColor="PWH"
      color="BLK"
      to="/"
    >
      Log out of {props.username}
    </SettingsButton>
  </React.Fragment>
);

export default SettingsButtons;
