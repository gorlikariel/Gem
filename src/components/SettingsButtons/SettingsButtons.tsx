import * as React from 'react';
import SettingsButton from './SettingsButton/SettingsButton';
import * as palette from '../../styleguide/theme';

interface SettingsButtonsProps {
  checked: boolean;
  logout: any;
  username: string;
}

const SettingsButtons: React.SFC<SettingsButtonsProps> = props => {
  const { checked, logout, username } = props;
  return (
    <>
      <SettingsButton
        in={checked}
        icon="accountIcon"
        bgImg={palette.G3}
        color={palette.PWH}
        to="/account-settings"
      >
        Account Settings
      </SettingsButton>
      <SettingsButton
        in={checked}
        timeout={400}
        icon="alarmIcon"
        bgImg={palette.G2}
        color={palette.PWH}
        to="/alarm-settings"
      >
        Alarm Settings
      </SettingsButton>
      <SettingsButton
        in={checked}
        timeout={600}
        icon="pillIcon"
        bgImg={palette.PG}
        color={palette.PWH}
        to="/pill-settings"
      >
        Pill Settings
      </SettingsButton>
      <SettingsButton
        in={checked}
        timeout={800}
        onClick={logout}
        bgImg={palette.PWH}
        color={palette.BLK}
        to="/"
      >
        Log out of {username}
      </SettingsButton>
    </>
  );
};
export default SettingsButtons;
