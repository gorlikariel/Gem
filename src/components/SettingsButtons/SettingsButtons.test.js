import React from 'react';
import { shallow, EnzymeAdapter, configure } from 'enzyme';
import SettingsButtons from './SettingsButtons';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';
import SettingsButton from './SettingsButton/SettingsButton';
Enzyme.configure({ adapter: new Adapter() });
describe('SettingsButtons', () => {
  it('should render 4 settings buttons, the last one being the logout button which renders username based on props.username', () => {
    const username = 'Test Username';
    const settingsButtons = shallow(
      <SettingsButtons checked={true} logout={() => null} username={username} />
    );
    expect(
      settingsButtons
        .find(SettingsButton)
        .last()
        .props().children[1] === username
    );
    const tree = toJson(settingsButtons);
    expect(tree).toMatchSnapshot();
  });
});
