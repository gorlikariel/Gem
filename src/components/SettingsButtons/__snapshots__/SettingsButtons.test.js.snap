// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsButtons Renders 4 settings buttons, the last one being the logout button which renders username based on props.username 1`] = `
<Fragment>
  <SettingsButton
    bgImg="linear-gradient(to right, #77b4e4, #4ce49e)"
    color="#ffffff"
    icon="accountIcon"
    in={true}
    to="/account-settings"
  >
    Account Settings
  </SettingsButton>
  <SettingsButton
    bgImg="linear-gradient(to right, #66bbf7, #6696fb)"
    color="#ffffff"
    icon="alarmIcon"
    in={true}
    timeout={400}
    to="/alarm-settings"
  >
    Alarm Settings
  </SettingsButton>
  <SettingsButton
    bgImg="linear-gradient(to right, #7c9dfb, #a388fa)"
    color="#ffffff"
    icon="pillIcon"
    in={true}
    timeout={600}
    to="/pill-settings"
  >
    Pill Settings
  </SettingsButton>
  <SettingsButton
    bgImg="#ffffff"
    color="black"
    in={true}
    onClick={[Function]}
    timeout={800}
    to="/"
  >
    Log out of 
    Test Username
  </SettingsButton>
</Fragment>
`;
